sdk/java/core/src/main/kotlin/com/keepersecurity/secretsManager/core/SecretsManager.kt

```diff
@@ -744,22 +768,18 @@ private fun fetchAndDecryptSecrets(
     if (response.records != null) {
         response.records.forEach {
             val recordKey = decrypt(it.recordKey, appKey)
-            val decryptedRecord = decryptRecord(it, recordKey)
-            if (decryptedRecord != null) {
-                records.add(decryptedRecord)
-            }
+            decryptRecord(it, recordKey).onSuccess { keeperRecord -> records.add(keeperRecord) }
```

Comment:
this is the magic part where if there's a failure it skips gracefully (i.e. `onSuccess` only gets called if we ge a successful `Result`)